src/components/PostList.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useState, useMemo } from 'react';
+import debounce from 'lodash/debounce';
+
+import { PostWithUser } from '../api/api';
+import Post from './Post';
+
+type Props = {
+  posts: PostWithUser[];
+}
+
+const PostList: React.FC<Props> = ({ posts }) => {
+  const [query, setQuery] = useState('');
+  const [filterQuery, setfilterQuery] = useState('');
+
+
+  const getVisiblePosts = (allPosts: PostWithUser[], searchQuery: string) => {
+    const pattern = new RegExp(searchQuery, 'i');
+
+    return allPosts.filter(({ body, title }) => pattern.test(body + title))
+  }
+
+  const vissiblePosts = useMemo(
+    () => getVisiblePosts(posts, filterQuery),
+
+    [filterQuery]
```

Comment:
```suggestion
    [filterQuery, posts]
```
if posts changed - you should recompute this value